core/src/banking_stage/consume_worker.rs

```diff
@@ -392,6 +393,10 @@ pub(crate) mod external {
                     bank,
                     &transactions,
```

Comment:
We need to do an additional check here for all_or_nothing; if any of the transactions fail to parse/sanitize/resolve, then it doesn't have an entry in `transactions`.

All you need to do is just check `translation_results.len() != transactions.len()`, since `translation_results` has an entry for every tx in the original message.